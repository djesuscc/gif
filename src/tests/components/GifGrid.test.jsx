import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test int component <GifGrid />', () => {
    const category = 'Dragon Ball';

    test('should show correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });
   test('should show items when the images are load with the useFetchGifs hook', () => {

        const gifs = [{
            id: 'abc',
            url: 'hhtps://localhost/test.png',
            title: 'Image test'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
   });
});