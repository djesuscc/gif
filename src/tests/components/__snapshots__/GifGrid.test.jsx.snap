// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test int component <GifGrid /> should show correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test int component <GifGrid /> should show items when the images are load with the useFetchGifs hook 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Image test"
      url="hhtps://localhost/test.png"
    />
  </div>
</Fragment>
`;
